CMakeLists.txt

```diff
@@ -151,16 +151,7 @@ ELSE()
   MARK_AS_ADVANCED(MYSQL_PROJECT_NAME)
 ENDIF()
 
-# CMake version 3.22.1 and above
-# Policy CMP0048 is not set: project() command manages VERSION variables.
-INCLUDE(mysql_version)
-IF(POLICY CMP0048)
-  CMAKE_POLICY(SET CMP0048 NEW)
-  PROJECT(${MYSQL_PROJECT_NAME}
-    VERSION ${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION})
-ELSE()
-  PROJECT(${MYSQL_PROJECT_NAME})
-ENDIF()
+PROJECT(${MYSQL_PROJECT_NAME})
```

Comment:
This may still result in warning on newer cmake versions. 

If we need to have WSREP_VERSION defined before `cmake/mysql_version.cmake`, we can do either of the below.

1. Move the version related cmake instructions to `wsrep_version.cmake` and include it before `mysql_version`
2. Set WSREP_VERSION in `cmake/mysql_version.cmake`. This way it will set wsrep version variables along with server version.
3. Move the check for `execvpe` to a different file and include `wsrep` and `mysql_version` before `PROJECT()`

Reply:
Done. Opted with alternative #2.

Reply:
Why we cant just move `INCLUDE(wsrep)` from line 305 (after comment `#Add macros`) before `INCLUDE(mysql_version)`

Reply:
wsrep.cmake contains _**CHECK_FUNCTION_EXISTS(execvpe HAVE_EXECVPE)**_ which requires C or CXX language enabled and compilation fails with the below message.

CMake Error at /usr/share/cmake-3.22/Modules/CheckFunctionExists.cmake:89 (message):
  CHECK_FUNCTION_EXISTS needs either C or CXX language enabled
Call Stack (most recent call first):
  cmake/wsrep.cmake:22 (CHECK_FUNCTION_EXISTS)
  CMakeLists.txt:156 (INCLUDE)